packages/@aws-cdk/aws-cloudwatch/lib/alarm.ts

```diff
@@ -39,6 +51,14 @@ export interface AlarmProps extends CreateAlarmOptions {
    * custom Metric objects by instantiating one.
    */
   readonly metric: IMetric;
+
+  /**
+   * AlarmState to build composite alarm expressions.
+   *
+   * @default - ALARM
+   */
+  readonly alarmState?: AlarmState;
```

Comment:
I think making this configurable will just make the usage sites more confusing (because if you see an alarm in an AlarmRule position you can't immediately know what it means anymore).

It just always just be `ALARM`.

Reply:
i'm divided on this, i think there are scenarios when we/folks alarm when X is in ALARM state AND Y is in OK state. such logic does exist in certain scenarios. by removing this parameter, would we be limiting customers?

Reply:
are you suggesting that people creating this via ```Alarm``` construct always get it in ```ALARM``` and people that want more complex states can git it via ```AlarmRule```, that makes sense.

Reply:
Yes, that's what I was suggesting.